spec/services/halls/update_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Halls::Update do
+  let!(:hall) { create :hall, name: 'name', capacity: 40 }
+  let(:params) do
+    {
+      name:,
+      capacity: 41
+    }
+  end
+  let(:name) { 'name updated' }
+  let(:instance) { described_class.new(hall.id, params) }
+
+  subject { instance.call }
+
+  describe '.call' do
+    it 'updates hall' do
+      expect { subject }
+        .to change { hall.reload.name }.from('name').to('name updated')
+        .and change { hall.reload.capacity }.from(40).to(41)
+    end
+
+    it 'retuns hall object' do
+      expect(subject).to eq(hall)
+    end
```

Comment:
```suggestion
    it 'retuns hall object' do
      expect(hall_update).to eq(hall)
    end
```

also you have some other shorter syntax if you declared subject

```suggestion
    it { is_expected.to eq(hall) }
```